src/components/report-suspicious-activity/report-suspicious-activity-controller.ts

```diff
@@ -0,0 +1,106 @@
+import { Request, Response, NextFunction } from "express";
+import { PATH_DATA } from "../../app.constants";
+import { formatEvent } from "../../utils/activityHistory";
+import { getAppEnv, getDynamoActivityLogStoreTableName } from "../../config";
+import { DynamoDB } from "aws-sdk";
+import { dynamoDBService } from "../../utils/dynamo";
+import { ActivityLogEntry, FormattedActivityLog } from "../../utils/types";
+import assert from "node:assert";
+import { logger } from "../../utils/logger";
+
+const activityLogDynamoDBRequest = (
+  subjectId: string,
+  eventId: string
+): DynamoDB.Types.QueryInput => ({
+  TableName: getDynamoActivityLogStoreTableName(),
+  KeyConditionExpression: "user_id = :user_id AND event_id = :event_id ",
+  ExpressionAttributeValues: {
+    ":user_id": { S: subjectId },
+    ":event_id": { S: eventId },
+  },
+});
+
+export async function reportSuspiciousActivityGet(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> {
+  try {
+    assert.ok(
+      req.query.event !== undefined && req.query.event !== "",
+      "mandatory query parameter 'event' missing from request"
+    );
+  } catch (err) {
+    req.log.error(err.message);
+    return next();
+  }
```

Comment:
If we used a route parameter, we wouldn't need this

Reply:
Same as for last comment.